### PR TITLE
make cloud-config the configuration mechanism for RancherOS

### PR DESCRIPTION
**Pull request by imikushin**
Do not merge yet. 

A very obscure bug (and now traced) was causing `ros config get` to 
print out YAML unmarshaling errors instead of configuration values.
`c.readArgs()` usage is now commented out (config/config.ro:228), and I need to 
code the proper fix now. 

Currently, it finally compiles and runs, but needs cleanup.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -12,27 +11,27 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
-func (c *Config) privilegedMerge(newConfig Config) error {
+func (c *RCloudConfig) privilegedMerge(newConfig RCloudConfig) error {
```
Comment (ibuildthecloud) [inline config/config.go]: Rename CloudConfig

**Comment by ibuildthecloud** [comment]
Lets delete `system_container` from config and just use `services`.  Also remove any "readonly" logic in config.go

File: config/types.go [hunk 2]
```diff
@@ -24,14 +27,17 @@ const (
 	RELOAD_CONFIG = "io.rancher.os.reloadconfig"
 	SCOPE         = "io.rancher.os.scope"
 	SYSTEM        = "system"
+
+	OsConfigFile          = "/os-config.yml"
+	CloudConfigFile       = "/var/lib/rancher/conf/cloud-config.yml"
+	CloudConfigScriptFile = "/var/lib/rancher/conf/cloud-config-script"
+	MetaDataFile          = "/var/lib/rancher/conf/metadata"
+	LocalConfigFile       = "/var/lib/rancher/conf/local-config.yml"
```
Comment (ibuildthecloud) [inline config/types.go]: rename to cloud-config-local.yml

**Comment by imikushin** [comment]
fixes #454

**Pull request merged**